### PR TITLE
Rust syntax highlighting v2

### PR DESCRIPTION
**Pull request by kirawi**
The aim is to make it more accurate by basing it entirely on the source `grammar.js`, and use less patterns by leveraging `locals.scm`, as well as updating it to use the new scopes.

### CODE REVIEW

**Comment by archseer** [comment]
> more accurate by basing it entirely on the source `grammar.js`

What do you mean here? The queries were already based on the grammar or they wouldn't match the tree.

**Comment by kirawi** [comment]
Before I got them from plugging in example code into the playground. This time I'm looking directly at the source.

File: runtime/queries/rust/highlights.scm [hunk 1]
```diff
@@ -311,46 +75,117 @@
...
+(const_item
+  name: (identifier) @constant)
+(static_item
+  name: (identifier) @constant)
+(function_item
+  name: (identifier) @function)
+(function_signature_item
+  name: (identifier) @function)
 
-"?" @special
```
Comment (kirawi) [inline runtime/queries/rust/highlights.scm:352]: `@special` no longer exists, what should this be changed to?

**Comment by archseer** [comment]
Please reopen if you ever revisit this PR

**Pull request closed without merging**